service/controller/legacy/v29/resource/ebsvolume/error.go

```diff
@@ -12,3 +12,12 @@ var invalidConfigError = &microerror.Error{
 func IsInvalidConfig(err error) bool {
 	return microerror.Cause(err) == invalidConfigError
 }
+
+var volumeAttachedError = &microerror.Error{
+	Kind: "volumeAttachedError",
+}
+
+// isVolumeAttached asserts volumeAttachedError.
+func isVolumeAttached(err error) bool {
```

Comment:
That does not make sense. When you test this code you will see that the resource will never be cancelled because the matcher matches against the `volumeAttachedError` of the `ebs` service package, which will never match. We do not need the error and matcher here in the resource. We only need to use the matcher of the `ebs` service in the resource. Also not that matcher should always be public as described by `fmt`.